src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -1,69 +1,86 @@
-        // In Part II:
-            // given a BDIO, call BdioToCLLComponentTransformer to get CLL input file (components-source.json)
-            // call library w/ CLL input
-            // return the resulting file (not anticipating any post-processing except saving it in the correct directory)
-        return generatePlaceHolderJsonFileForNow(scanOutputFolder);
+    public void locateComponentsForNonPersistentOnlineDetectorScan(List<DeveloperScansScanView> rapidFullResults, File scanOutputFolder, File sourceDir) {
+        List<Component> componentsList = scanResultTransformer.transformScanResultToComponentList(rapidFullResults);
+        Input componentLocatorInput = generateComponentLocatorInput(componentsList, sourceDir);
+        callComponentLocatorObfuscatedJar(componentLocatorInput, scanOutputFolder);
     }
 
     /**
-     * Placeholder file for testing purposes, saves a JSON file in the appropriate directory with the appropriate name.
-     * @param scanOutputFolder
-     * @return
+     * Given a BDIO, generates an output file consisting of the list of unique components detected and their declaration
+     * locations.
+     * @param bdio from running offline Detector scan
+     * @param scanOutputFolder Detect's output subdirectory where this file will be saved
+     * @param sourceDir source directory of project being scanned
      * @throws DetectUserFriendlyException
      */
-    private static File generatePlaceHolderJsonFileForNow(File scanOutputFolder) throws DetectUserFriendlyException {
+    public void locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder, File sourceDir) {
+        List<Component> componentsList = bdioTransformer.transformBdioToComponentList(bdio);
+        Input componentLocatorInput = generateComponentLocatorInput(componentsList, sourceDir);
+        callComponentLocatorObfuscatedJar(componentLocatorInput, scanOutputFolder);
+    }
+
+    private Input generateComponentLocatorInput(List<Component> componentsList, File sourceDir) {
+        return new Input(sourceDir.getAbsolutePath(), new Metadata(), componentsList);
+    }
+
+    private void callComponentLocatorObfuscatedJar(Input componentLocatorInputObj, File scanOutputFolder) {
+        try {
+            int status = Application.locateComponents(componentLocatorInputObj, scanOutputFolder.toString() + DETECT_OUTPUT_FILE_NAME);
+            if (status != 0) {
+                logger.info("There was a problem during Component Locator execution. Failed to generate Component Location Analysis file.");
+            }
+        } catch (Exception e) {
+            logger.info("There was a problem during Component Locator execution. Failed to generate Component Location Analysis file.");
+        }
+    }
+
+    private File serializeInputToJson(File saveInputFileDir, Input libInput) throws DetectUserFriendlyException {
```

Comment:
Leaving this method here for dev use, will be removed before feature branch is merged to master.